### DISCUSSION TITLE
Expose command and mode in config script

### DISCUSSION THREAD

**Discussion by 0x241F31** [withastro/roadmap#352] [Proposal]
I'm migrating from Vite, it has ability to export function from config with following arguments:
```ts
export declare interface ConfigEnv {
    command: 'build' | 'serve';
    mode: string;
    /**
     * @experimental
     */
    ssrBuild?: boolean;
}
```
I want to push cssnano plugin to Vite config if command is 'build'. How can I achieve the same in Astro?
I see integration can access it, but `astro.config.mjs` disallows exporting function. Also, import.meta.env is irrelevant at config time.
![image](https://user-images.githubusercontent.com/17229619/197149091-1f0b720f-634a-4088-b952-876ef6849db7.png)

I'm using current workaround `const dev = process.env.npm_lifecycle_event === 'dev'`, as I don't want to mess with env variables and packages like `cross-env`

**Comment by delucis** [comment]
Hi @MrFoxPro, I think you're on the right track!

You should be able to push that plug-in using an integration that provides the `astro:config:setup` hook. Something like the following:

```js
// astro.config.mjs

defineConfig({
  integrations: [
    {
      name: 'my-cssnano-integration',
      'astro:config:setup': ({
        command,
        updateConfig,
      }) => {
        if (command === 'build') {
          updateConfig({
            vite: {
              plugins: [cssnano],
            },
          });
        }
      },
    }
  ],
});
```

I'm not entirely sure how the cssnano plug-in is used but something like the above updated for how you pass cssnano to Vite might work!

**Comment by florian-lefebvre** [comment]
Closing because this is outdated
